File: frontend/src/api/estoque.ts
import api from "./client"
import type { Produto } from "./produtos"

export interface EstoqueItem {
  produto_id: number
  quantidade: number
  quantidade_minima: number
  produto: Produto
}

export const getEstoque = () => api.get<EstoqueItem[]>("/estoque/").then(r => r.data)
export const atualizarEstoque = (produto_id: number, dados: { quantidade: number; quantidade_minima?: number }) =>
  api.patch<EstoqueItem>(`/estoque/${produto_id}`, dados).then(r => r.data)

File: backend/app/models/estoque.py
from sqlalchemy import Column, ForeignKey, Integer
from sqlalchemy.orm import relationship

from app.database import Base


class EstoqueCentral(Base):
    __tablename__ = "estoque_central"

    id = Column(Integer, primary_key=True, index=True)
    produto_id = Column(Integer, ForeignKey("produtos.id"), unique=True, nullable=False)
    quantidade = Column(Integer, default=0)
    quantidade_minima = Column(Integer, default=5)

    produto = relationship("Produto", back_populates="estoque")
